src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,65 @@ public function request(string $method, string $url, array $options = []): Respo
             $curlopts[\CURLOPT_TIMEOUT_MS] = 1000 * $options['max_duration'];
         }
 
+        if (isset($options['extra']['curl']) && \is_array($options['extra']['curl'])) {
+            $extraCurlopts = array_keys($options['extra']['curl']);
+            $curloptsToConfig = [
+                \CURLOPT_HTTPAUTH => 'auth_ntlm',
+                \CURLOPT_USERPWD => 'auth_ntlm',
+                \CURLOPT_RESOLVE => 'resolve',
+                \CURLOPT_NOSIGNAL => 'timeout',
+                \CURLOPT_HTTPHEADER => 'headers',
+                \CURLOPT_INFILE => 'body',
+                \CURLOPT_READFUNCTION => 'body',
+                \CURLOPT_INFILESIZE => 'body',
+                \CURLOPT_POSTFIELDS => 'body',
+                \CURLOPT_UPLOAD => 'body',
+                \CURLOPT_PINNEDPUBLICKEY => 'peer_fingerprint',
+                \CURLOPT_UNIX_SOCKET_PATH => 'bindto',
+                \CURLOPT_INTERFACE => 'bindto',
+                \CURLOPT_TIMEOUT_MS => 'max_duration',
+                \CURLOPT_MAXREDIRS => 'max_redirects',
+                \CURLOPT_PROXY => 'proxy',
+                \CURLOPT_NOPROXY => 'no_proxy',
+                \CURLOPT_SSL_VERIFYPEER => 'verify_peer',
+                \CURLOPT_SSL_VERIFYHOST => 'verify_host',
+                \CURLOPT_CAINFO => 'cafile',
+                \CURLOPT_CAPATH => 'capath',
+                \CURLOPT_SSL_CIPHER_LIST => 'ciphers',
+                \CURLOPT_SSLCERT => 'local_cert',
+                \CURLOPT_SSLKEY => 'local_pk',
+                \CURLOPT_KEYPASSWD => 'passphrase',
+                \CURLOPT_CERTINFO => 'capture_peer_cert_chain',
+                \CURLOPT_USERAGENT => 'normalized_headers',
+                \CURLOPT_REFERER => 'headers',
```

Comment:
can you add the options used in CurlResponse please?

Reply:
Added.

Reply:
a few more:
CURLOPT_RETURNTRANSFER
CURLOPT_URL
CURLOPT_FOLLOWLOCATION
CURLOPT_HEADER
CURLOPT_NOBODY
CURLOPT_CONNECTTIMEOUT
CURLOPT_HEADEROPT
CURLOPT_HTTP_VERSION
CURLOPT_UPLOAD <= eg this one is already found in the source, dunno why you didn't list it. Can you double check if there are any others of this kind?

Reply:
CURLOPT_UPLOAD check was introduced in https://github.com/symfony/symfony/pull/38026/commits/9e1ce25347478495c2387b2ab90ff2caa8e1db81#diff-2548b043c0f673f37f416b4b6f3bcbc7R289 

Also, found `CURLOPT_PROTOCOLS`, `CURLOPT_REDIR_PROTOCOLS` and `CURLOPT_COOKIEFILE`, that are defined inside `CurlHttpClient` and not checked for presence in extra.curl.